### PR TITLE
docs: add scaling guide for crawlers

### PR DESCRIPTION
**Pull request by vladfrangu**
Would love feedback on wording and if I explained everything well enough! I covered the more important options first (as well as added a shortcut for `maxTasksPerMinute` at the top level of the crawler options for visibility)

Google Chrome - 2022-07-07 at 17 39 23@2x

### CODE REVIEW

**Comment by B4nan** [comment]
cc @metalwarrior665 @mnmkng @petrpatek

File: docs/guides/session_management.mdx [hunk 1]
```diff
@@ -15,7 +15,7 @@ import PlaywrightSource from '!!raw-loader!./session_management_playwright.ts';
 import PuppeteerSource from '!!raw-loader!./session_management_puppeteer.ts';
 import StandaloneSource from '!!raw-loader!./session_management_standalone.ts';
 
-&#8203;<ApiLink to="core/class/SessionPool">`SessionPool`</ApiLink> is a class that allows us to handle the rotation of proxy IP addresses along with cookies and other custom settings in Crawlee.
+​<ApiLink to="core/class/SessionPool">`SessionPool`</ApiLink> is a class that allows us to handle the rotation of proxy IP addresses along with cookies and other custom settings in Crawlee.
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply docs/guides/session_management.mdx:18]: Ahhh yes, I didn't think this through for reviewers. I replaced that html code with a 0-width space (the html code itself rendered as raw text on the page where it showed them all

[the remark this answers is not in the document]
Comment (vladfrangu) [reply docs/guides/session_management.mdx:18]: It is but it renders the exact text in the guides page (where you see all the squares with available guides)

File: docs/guides/scaling_crawlers.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+## `minConcurrency` and `maxConcurrency`
+
+These control how many parallel requests can be run at a time. By default, crawlers will start with one parallel request at a time and scale up over time to a maximum of 1000 requests at a time.
+
+:::caution Don't set `minConcurrency` too high!
+
+Setting this option too high compared to the available system resources will make your crawler run extremely slow or might even crash.
+
+We recommend you leave it at the default value we set and let the crawlers scale up and down automatically based on available resources instead.
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply docs/guides/scaling_crawlers.mdx]: Rewording this sentence to made a distinction between "we" the guide reader and "we" crawlee was fun... could you check it now and tell me if it's better? 😅

[the remark this answers is not in the document]
Comment (AndreyBykov) [reply docs/guides/scaling_crawlers.mdx]: I guess should be good :) And tbh, I still have several places where I also need to do the same thing :)

File: docs/guides/scaling_crawlers.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+It's recommended to leave it at the default value that is provided and letting the crawler scale up and down automatically based on available resources instead.
+
+:::
+
+<CodeBlock language="js">
+	{MinMaxConcurrencySource}
+</CodeBlock>
+
+## `maxRequestsPerMinute`
```
Comment (mnmkng) [inline docs/guides/scaling_crawlers.mdx:38]: I would place this above the min/maxConcurrency as it's a much better rate limiting option than those two.

I'm also thinking that maybe we should set some reasonable default, like 200 or something that does not overload a website 😄

Comment (B4nan) [reply docs/guides/scaling_crawlers.mdx:38]: Makes sense. On the other hand, if we have `maxConcurrency` set to 1000 as the paragraph above states, maybe we should adjust that as well?

File: docs/guides/scaling_crawlers.mdx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	{MaxRequestsPerMinuteSource}
+</CodeBlock>
+
+## Advanced configurations
+
+While the options above should be enough for most users, if we wanted to get super deep into the configuration of the autoscaling pool (the internal utility in Crawlee that helps us allow crawlers to scale up and down), we can do so through the <ApiLink to="cheerio-crawler/interface/CheerioCrawlerOptions#autoscaledPoolOptions">`autoscaledPoolOptions`</ApiLink> object available on crawler options.
+
+:::danger Complex options up ahead!
+
+This section is super advanced and, unless we test the changes extensively and know what we're doing, it's better to leave these options to the default configurations provided, as they are most likely going to work for us without much fuss.
```
Comment (B4nan) [inline docs/guides/scaling_crawlers.mdx:52]: while generally i want to see the "we" in docs, here (or in general in hints like this) i'd say its better to keep using the "you". as the docs should be about telling a story, but hints are like "hey you, heads up!"

```suggestion
This section is super advanced and, unless you test the changes extensively and know what you're doing, it's better to leave these options to their defaults, as they are most likely going to work fine without much fuss.
```

File: docs/guides/scaling_crawlers.mdx (hunk 3 above)
Comment (vladfrangu) [reply docs/guides/scaling_crawlers.mdx:38]: We could change the default for maxConcurrency to 200 from 1000, I think that's a smart idea. I'll also move this above them as I agree with you

File: docs/guides/scaling_crawlers.mdx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+We can think of this as the point where the autoscaling pool can attempt to scale up (or down), monitor if there's any changes, and correct them if necessary.
+
+### `scaleUpStepRatio` and `scaleDownStepRatio`
+
+These values define the fractional amount of desired concurrency to be added or subtracted as the autoscaling pool scales up or down. Both of these values default to `0.05`.
+
+:::caution
+
+If we're planning on modifying these, make sure the value we set is not `0`, and that it is positive!
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply docs/guides/scaling_crawlers.mdx:78]: We don't actually have strict validation for this... 😓  Would probably help to fix that too

File: docs/guides/avoid_blocking.mdx [hunk 6]
```diff
@@ -1,28 +1,30 @@
...
 Browser fingerprint is a collection of browser attributes and significant features that can show if our browser is a bot or a real user. Moreover, even most browsers have these unique features that allow the website to track the browser even within different IP addresses. This is the main reason why scrapers should change browser fingerprints while doing browser-based scraping. In addition, it should reduce blocking significantly.
 
 Changing browser fingerprints can be a tedious job. Luckily, Crawlee provides this feature out of the box with zero configuration necessary. Let's take a look at how it is done.
 
- Changing browser fingerprints is available in `PlaywrightCrawler` and `PuppeteerCrawler`. We have to pass the `useFingerprints` option to the `browserPoolOptions`.
+Changing browser fingerprints is available in `PlaywrightCrawler` and `PuppeteerCrawler`. We have to pass the `useFingerprints` option to the `browserPoolOptions`.
```
[the remark this answers is not in the document]
Comment (AndreyBykov) [reply docs/guides/avoid_blocking.mdx:12]: I updated it, but it was more than a month ago, and it was probably one of the first ones I updated. And although it's technically correct, I agree that it's worth adding that `true` is a default option and now we would rather switch it off. I would either rebase my current PR or just include it with the next one with last changes.

**Review by B4nan — APPROVED** [review]

**Pull request merged**